---
title: "Investigating Influenza Susceptibility by Population and Sex"
subtitle: UPGG Computational Bootcamp Final Project
author: Angela Jones
output:
  html_document:
    toc: true 
    toc_float: 
      collapsed: false 
      smooth_scroll: false 
---

# Setup

## Load libraries
***

```{r setup, include=FALSE}
knitr::opts_chunk$set(fig.width=15, fig.height=6)

library(tidyverse) # includes dplyr and ggplot2
library(magrittr) # allows for double piping
library(stringr) # string manipulation
library(ggpubr) # extra features for ggplot2
library(ggsignif) # add significance tests to ggplot2
library(ggsci) # cool color palettes
library(fontawesome) # cool icons!
library(cowplot) # plot multiple graphs at once
```


# The Dataset `r fa("viruses")`

This dataset was generated by the Ko lab through a new single-cell RNAseq protocol called **scHi-HOST**. In scHi-HOST, human lymphoblastoid cell lines from 48 individuals (24 females, 24 males) across three distinct populations are infected with influenza A virus. Single-cell RNA-seq of the infected cells and downstream bioinformatics allows for the deconvolution of the viral and host RNA to create net flu phenotypes per individual (_e.g._ median flu reads). This protocol is a recent improvement on the previously published [Hi-HOST](http://h2p2.oit.duke.edu/H2P2Home/) system.

## Load and investigate dataset through `glimpse`

```{r glimpse, echo=FALSE, message=FALSE}
flu_phenotypes <- read_csv("G:/My Drive/AngelaJones2021/Data_analysis/archive/PR8_Phenotypes/PR8ViralBurden_120720.csv") # load dataset from Google Drive

glimpse(flu_phenotypes) # check dimensions and column types of dataset
```
## Remove extra columns from dataset and tidy 'Population' column

```{r tidy, echo=FALSE}
flu_phenotype_trimmed <- flu_phenotypes %>% # remove extra cols
  select(-(IID_ID:MOT_ID))
names(flu_phenotype_trimmed)[1] <- "FAM_ID" # rename first col
flu_phenotype_trimmed %<>% mutate(Population = str_to_title(Population)) # change text format of populations
glimpse(flu_phenotype_trimmed) # see how it looks after tidy
```

***

# Analyze Dataset `r fa("search")`

First, investigate if different individuals have more flu reads per cell than others. We can even start to compare by population here.

```{r bar-graph, echo = FALSE}
p1 <- flu_phenotype_trimmed %>%  # plot bar graph of flu reads by individual
  ggplot(aes(x = FAM_ID, y = log2MedianNormViralReads)) +
  geom_col(color = "darkgray", fill = "darkgray", alpha = 0.6) +
  scale_x_discrete(guide = guide_axis(angle = 45)) +
  theme_pubr() +
  theme(axis.text.x = element_text(size = 8)) +
  labs(x = "Individual", y = "log2[Median Viral Reads]", title = "Individual variation in scHi-HOST viral reads")

p2 <- flu_phenotype_trimmed %>% # plot bar graph of flu reads by individual - colored by population
  ggplot(aes(x = FAM_ID, y = log2MedianNormViralReads, color = Population, fill = Population)) +
  geom_col(alpha = 0.6) +
  scale_x_discrete(guide = guide_axis(angle = 45)) +
  scale_color_npg() +
  scale_fill_npg() +
  theme_pubr() +
  theme(axis.text.x = element_text(size = 8)) +
  labs(x = "Individual", y = "log2[Median Viral Reads]", title = "Individual variation in scHi-HOST viral reads by population")

plot_grid(p1,p2, ncol = 2) # use cowplot to plot both graphs at once

```

It looks like there is a lot of variation between individuals and maybe trends by population, but that's difficult to visualize with the above plots. Another method of visualization will be helpful.

```{r box-plot, echo=FALSE}
p3 <- flu_phenotype_trimmed %>% # box plot of flu burden by population
  ggplot(aes(x = Population, y = log2MedianNormViralReads, fill = Population, color = Population)) +
  geom_boxplot(outlier.shape = NA, alpha = 0.6) +
  geom_point(size=3, position = position_jitterdodge(jitter.width = 0.1), alpha = 0.8) +
  scale_color_npg() +
  scale_fill_npg() +
  geom_signif(comparisons = list(c("British", "Luhya"),c("Luhya",'Southern Han Chinese'), c("British", "Southern Han Chinese")), 
              map_signif_level=FALSE, color = "black", step_increase = 0.07, tip_length = 0) +
  labs(y = "IAV log2[Median Viral Reads]", x = "Population", title = "Luhya population shows highest viral burden") +
  theme_pubr()
p3
```

The above plot is much more useful. From this plot we can see that the [Luhya](https://en.wikipedia.org/wiki/Luhya_people) population (in Kenya) appears to more susceptible to flu infection than the British and Southern Han Chinese populations.

While population differences are useful to know, my research is focused on sex differences in flu susceptibility and how that differs between populations.

```{r box-plot2, echo = FALSE}
p4 <- flu_phenotype_trimmed %>% # box plot of flu burden by sex (all individuals)
  ggplot(aes(x = Sex, y = log2MedianNormViralReads, fill = Sex, color = Sex)) +
  geom_boxplot(outlier.shape = NA, alpha = 0.6) +
  geom_point(size=3, position = position_jitterdodge(jitter.width = 0.1), alpha = 0.8) +
  scale_color_manual(values=c("#CC5500", "#3A84A6")) +
  scale_fill_manual(values=c("#CC5500", "#3A84A6")) +
  stat_compare_means(method = "t.test", label.x.npc = 0.4, label.y.npc = 0.9) +
  labs(y = "IAV log2[Median Viral Reads]", x = "Population", title = "Males show higher viral burden") +
  theme_pubr()

p5 <- flu_phenotype_trimmed %>% # box plot of flu burden by sex in each population
  ggplot(aes(x = Population, y = log2MedianNormViralReads, fill = Sex, color = Sex)) +
  geom_boxplot(outlier.shape = NA, alpha = 0.6) +
  geom_point(size=3, position = position_jitterdodge(jitter.width = 0.1), alpha = 0.8) +
  scale_color_manual(values=c("#CC5500", "#3A84A6")) +
  scale_fill_manual(values=c("#CC5500", "#3A84A6")) +
  stat_compare_means(method = "t.test", label.x.npc = 0.4, label.y.npc = 0.9) +
  labs(y = "IAV log2[Median Viral Reads]", x = "Population", title = "Luhya population shows largest sex difference in viral burden") +
  theme_pubr()

plot_grid(p4,p5, ncol = 2) # use cowplot to plot both figures
```

***

# Conclusions `r fa("check-circle")`

The above analyses show that there are individual, population, and sex differences in influenza A susceptibility. The Luhya population appears to be more susceptible to infection than the other populations, although this difference is not statistically significant^[Student's T-Test]. When sex differences are considered, there is a trend towards males being more susceptible to flu infection than females^[p = 0.096, Student's T-Test]. Sex differences within each population are less robust, but the Luhya population shows the strongest sex difference in susceptibility with higher flu burden in males than females^[p = 0.17, Student's T-Test].

***

# Future Directions `r fa("binoculars")`

There appears to be sex differences across all populations as well as between populations. In order to understand the genetic mechanisms underlying these effects, I can investigate differential gene expression by sex and by population.

***

# Acknowledgements `r fa("hand-spock")`

I would like to thank Ben Schott and Liuyang Wang for generating this dataset, as well as the wonderful instructores for this bootcamp. Lastly, thank you all for the fun and collaborative experience of this module! Live long and prosper!
